File: app/src/main/java/com/victoria/bleled/data/db/UserDao.java
package com.victoria.bleled.data.db;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import com.victoria.bleled.data.model.ModelUser;

import java.util.List;

@Dao
public interface UserDao {
    @Query("SELECT * FROM tb_user")
    List<ModelUser> getAll();

    @Query("SELECT * FROM tb_user WHERE uid IN (:userIds)")
    List<ModelUser> loadAllByIds(int[] userIds);

    @Query("SELECT * FROM tb_user WHERE first_name LIKE :first AND " +
            "last_name LIKE :last LIMIT 1")
    ModelUser findByName(String first, String last);

    @Insert
    void insertAll(ModelUser... users);

    @Delete
    void delete(ModelUser user);
}

File: app/src/main/java/com/victoria/bleled/data/db/UserDao.kt
package com.victoria.bleled.data.db

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import com.victoria.bleled.data.model.ModelUser

@Dao
interface UserDao {
    @get:Query("SELECT * FROM tb_user")
    val all: List<ModelUser?>?

    @Query("SELECT * FROM tb_user WHERE uid IN (:userIds)")
    fun loadAllByIds(userIds: IntArray?): List<ModelUser?>?

    @Query(
        "SELECT * FROM tb_user WHERE name LIKE :first LIMIT 1"
    )
    fun findByName(first: String?): ModelUser?

    @Insert
    fun insertAll(vararg users: ModelUser?)

    @Delete
    fun delete(user: ModelUser?)
}